src/App.tsx

```diff
@@ -16,35 +17,98 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  enum SortType {
+    ALPHABET = 'alphabet',
+    LENGTH = 'length',
+    RESET = 'reset',
+  }
+
+  const visibleGoods = [...goodsFromServer].sort(
+    (good1: string, good2: string) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        case SortType.RESET:
+          return 0;
+
+        default:
+          return 0;
+      }
+    },
```

Comment:
Directly manipulating the array of goods from the server for sorting can be considered as interacting with the DOM directly. It is better to handle this logic inside a useEffect hook or a memoized value to ensure that React's state management is being used correctly. [CODE KNOWLEDGE]